# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: xpcom/io/FilePreferences.h
# Commit: ef46ec074d6a
# Full Hash: ef46ec074d6aba10d6e4342e305f36ead6996ddb
# Author: Honza Bambas <honzab.moz@firemni.cz>
# Date: 2018-06-07 21:50:48
# Regressor Bug: 1413868
# File Overlap Count: 1
# Description:
#   Bug 1413868. r=valentin
# ==============================================================================

diff -r 14f7dab3a9bd -r ef46ec074d6a xpcom/io/FilePreferences.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/xpcom/io/FilePreferences.h	Thu Jun 07 13:56:16 2018 -0400
@@ -0,0 +1,25 @@
+/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set ts=8 sts=2 et sw=2 tw=80: */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "nsIObserver.h"
+
+namespace mozilla {
+namespace FilePreferences {
+
+void InitPrefs();
+void InitDirectoriesWhitelist();
+bool IsBlockedUNCPath(const nsAString& aFilePath);
+
+namespace testing {
+
+void SetBlockUNCPaths(bool aBlock);
+void AddDirectoryToWhitelist(nsAString const& aPath);
+bool NormalizePath(nsAString const & aPath, nsAString & aNormalized);
+
+}
+
+} // FilePreferences
+} // mozilla